src/extension.ts

```diff
@@ -96,19 +115,178 @@ export const DOCUMENTATION_VERSION_PATHS = new Map<string, string>([
+
   completionProviderInstance.updateFhirEntities();
   tasks.registerTaskProvider('fsh', new SushiBuildTaskProvider());
-  return { definitionProviderInstance, completionProviderInstance };
+  return { definitionProviderInstance, completionProviderInstance, conversionProviderInstance };
+}
+
+export function deactivate() {
+  fhirFSH.dispose();
+}
+
+export async function conversionFSHtoFHIR(...file: any[]): Promise<void> {
+  fhirFSH.clear();
+  const fileUri: Uri = file[0];
+
+  const fshNames = await findNamesInFSHResource(fileUri);
+  fshNames.forEach(name => {
+    fhirFSH.appendLine('Found FSH resource in source: ' + name);
+  });
+
+  const sushiConfigInfo = await findConfiguration(fileUri, fhirFSH);
+
+  let fshResourcesToConvert: string[];
+
+  // If there is a sushi-config there is a project with multiple FSH files (including the one we are converting)
+  if (sushiConfigInfo.sushiconfig == null) {
+    const fshContent = await workspace.fs.readFile(fileUri);
+    const decoder = new TextDecoder();
+    const fshString = decoder.decode(fshContent);
+    fshResourcesToConvert = [fshString];
+  } else {
+    fshResourcesToConvert = await findMatchingFSHResourcesForProject(sushiConfigInfo.sushiconfig);
+  }
+
+  const fshtoFHIRDependencies: fhirtypes.ImplementationGuideDependsOn[] = [];
+
+  sushiConfigInfo.dependencies.forEach(dependency => {
+    const newDependency: fhirtypes.ImplementationGuideDependsOn = {
+      packageId: dependency.split('@')[0],
+      version: dependency.split('@')[1]
+    };
+    fshtoFHIRDependencies.push(newDependency);
+  });
+
+  const dependenciesParameter =
+    fshtoFHIRDependencies.length === 0 ? undefined : fshtoFHIRDependencies;
+
+  fhirFSH.appendLine('Converting FSH to FHIR...');
+  sushiClient
+    .fshToFhir(fshResourcesToConvert, {
+      canonical: sushiConfigInfo.canonical,
+      fhirVersion: sushiConfigInfo.version,
+      dependencies: dependenciesParameter
+    })
+    .then(result => {
+      result.errors.forEach(error => {
+        fhirFSH.appendLine('Error: ' + error.message);
+      });
+
+      result.warnings.forEach(warning => {
+        fhirFSH.appendLine('Warning: ' + warning.message);
+      });
+
+      findJsonResourcesInResult(result.fhir, fshNames).forEach(jsonResource => {
+        const uri = createJSONURIfromIdentifier(JSON.parse(jsonResource).id);
+        conversionProviderInstance.updated(jsonResource, uri);
+
+        workspace.openTextDocument(uri).then(doc => {
+          window.showTextDocument(doc, { preview: false, viewColumn: ViewColumn.Active });
+        });
+      });
+    })
+    .catch(error => {
+      fhirFSH.appendLine('Error: ' + error.message);
+    })
+    .finally(() => {
+      fhirFSH.appendLine('Finished!');
+    });
+
+  fhirFSH.show();
+}
+
+export async function conversionFHIRtoFSH(...file: any[]): Promise<void> {
+  fhirFSH.clear();
+
+  const fileUri: Uri = file[0];
+  const fhirObjects = gofshUtils.readJSONorXML(fileUri.path);
```

Comment:
I consistently received errors here when converting FHIR to FSH. On my (windows) system, the value of `fileUri.path` would be something like `/c:/some/path/StructureDefinition-MyObservation.json`. This would get resolved relative to my C drive, resulting in an error like this:
```
[error] Error: ENOENT: no such file or directory, open 'C:\c:\some\path\StructureDefinition-MyObservation.json'
```
The file seems to load correctly if the leading `/` is removed. This is a Windows-specific issue, so if you don't have a Windows machine available to test with, I can try to come up with a fix for this.

Reply:
using fsPath instead of path should make it universal (Unix and linux). For me this seems to work. Can you please verify this?

Reply:
Using `fsPath` indeed makes it work on my Windows system. Thank you for suggesting that!